evm/utils.js

```diff
@@ -644,153 +638,190 @@ function wasEventEmitted(receipt, contract, eventName) {
+            .map((promiseResult, index) => [promisedChainsResults[index].chainId, promiseResult.value.result]);
+
+        const failedResults = resultsWithErrLogs
+            .filter(
+                (promiseResult, _index) =>
+                    promiseResult.status === 'rejected' || (promiseResult.status === 'fulfilled' && promiseResult.value.loggerError),
+            )
+            .map((promiseResult, index) => {
+                const chainId = promisedChainsResults[index].chainId;
+                if (promiseResult.status === 'rejected') {
+                    return [chainId, promiseResult.reason.message];
+                } else {
+                    return [chainId, promiseResult.value.loggerError];
+                }
+            });
+
+        results = Object.fromEntries(successfulResults);
+        failedChains = Object.fromEntries(failedResults);
     }
 
-    const chainsDeepCopy = deepCopy(config.chains);
+    for (const [chainId, loggerError] of Object.entries(failedChains)) {
+        printError(`Failed with error on ${chainId}: ${loggerError}`);
+    }
 
-    let results = [];
-    for (const chainName of chains) {
-        const chain = config.chains[chainName.toLowerCase()];
+    printInfo(
+        'Succeeded chains',
+        `[ ${chains
+            .filter((chain) => !failedChains[chain.axelarId])
+            .map((chain) => chain.name)
+            .join(', ')} ]`,
+    );
 
-        if (chainsToSkip.includes(chain.name.toLowerCase()) || chain.status === 'deactive') {
-            printWarn('Skipping chain', chain.name);
-            continue;
-        }
+    printInfo(
+        'Failed chains',
+        `[ ${chains
+            .filter((chain) => failedChains[chain.axelarId])
+            .map((chain) => chain.name)
+            .join(', ')} ]`,
+    );
 
-        console.log('');
-        printInfo('Chain', chain.name, chalk.cyan);
+    if (save) {
+        saveConfig(config, options.env);
+    }
 
-        try {
-            const result = await processCommand(config.axelar, chain, chainsDeepCopy, options);
+    return results;
+};
 
-            if (result) {
-                results.push(result);
-            }
-        } catch (error) {
-            printError(`Failed with error on ${chain.name}`, error.message);
+const asyncChainTask = (processCommand, axelar, chain, chains, options) => {
+    let loggerOutput = '';
+    let loggerError = '';
+    let result;
 
-            if (!catchErr && !options.ignoreError) {
-                throw error;
-            }
-        }
+    const stdStream = new Writable({
+        write(chunk, _encoding, callback) {
+            loggerOutput += chunk.toString();
+            callback();
+        },
+    });
+    const errorStream = new Writable({
+        write(chunk, _encoding, callback) {
+            loggerError += chunk.toString();
+            callback();
+        },
+    });
+    return asyncLocalLoggerStorage.run({ stdStream, errorStream }, async () => {
```

Comment:
not very readable to have a complicated anonymous function defined during the return statement, define it earlier?